packages/react-composites/tests/browser/common/chatTestHelpers.ts

```diff
@@ -86,25 +87,57 @@ export const waitForMessageWithContent = async (page: Page, messageContent: stri
   await waitForSelector(page, `.ui-chat__message__content :text("${messageContent}")`);
 };
 
-export const waitForTypingIndicatorHidden = async (page: Page): Promise<void> => {
+/**
+ * Wait for typing indicators to appear and then dismiss them before continuing.
+ *
+ * Only select typing indicators under the targeted root node. By default, this means anywhere in the <body>.
+ */
+export const waitForAndHideTypingIndicator = async (page: Page, rootSelector = 'body'): Promise<void> => {
   await page.bringToFront();
-  await page.waitForTimeout(1000); // ensure typing indicator has had time to appear
-  const typingIndicator = await page.$(dataUiId(IDS.typingIndicator));
-  typingIndicator && (await typingIndicator.waitForElementState('hidden')); // ensure typing indicator has now disappeared
+  const indicator = await (await page.locator(rootSelector)).locator(dataUiId(IDS.typingIndicator));
+  // First make sure that the typing indicator appears.
+  await indicator.waitFor({
+    state: 'visible',
+    timeout: perStepLocalTimeout()
+  });
+  // Advance time to make the typing indicator disappear.
+  await page.evaluate(() => {
+    const currentDate = new Date();
+    // Typing indicator internal timeout is 10 seconds, so advance by
+    // a larger number.
+    currentDate.setSeconds(currentDate.getSeconds() + 11);
+    Date.now = () => currentDate.getTime();
+  });
+  // Now wait for the indicator to disappear.
+  await indicator.waitFor({
+    state: 'hidden',
+    timeout: perStepLocalTimeout()
+  });
 };
 
 export const waitForNSeenMessages = async (page: Page, n: number): Promise<void> => {
+  return waitForNOf(page, n, '[data-ui-status="seen"]', 'body');
```

Comment:
Note that before this PR, `waitForNSeenMessages` was simply broken: It would always return a `Promise`, which is truthy.

So this async function never really waited for the N messages to be seen.
In fact, its use was also broken -- it was never possible for the N messages to be seen with the chat pane closed.

So... two wrongs make a (flaky!) right. We just kept getting lucky that executing this selector in the browser took long enough that the messages were actually delivered.